modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts

```diff
@@ -40,14 +40,71 @@ const loadSkin = (isInline: boolean, editor: Editor): Promise<void> => {
   // Seems to work without, but adding a note in case things break later
   if (!Options.isSkinDisabled(editor) && Type.isString(skinUrl)) {
     return Promise.all([
-      loadUiSkins(editor, skinUrl),
-      loadShadowDomUiSkins(editor, skinUrl)
-    ]).then(SkinLoaded.fireSkinLoaded(editor), SkinLoaded.fireSkinLoadError(editor, 'Skin could not be loaded'));
+      loadUrlUiSkins(editor, skinUrl),
+      loadShadowDomUrlUiSkins(editor, skinUrl)
+    ]);
   } else {
-    return Promise.resolve(SkinLoaded.fireSkinLoaded(editor)());
+    return Promise.resolve();
   }
 };
 
+const loadRawCss = (editor: Editor, key: string, css: string, styleSheetLoader: StyleSheetLoader): void => {
+  // Ensure the stylesheet is cleaned up when the editor is destroyed
+  editor.on('remove', () => styleSheetLoader.unloadRawCss(key));
+  return styleSheetLoader.loadRawCss(key, css);
+};
+
+const loadRawUiSkins = (editor: Editor, skinKey: string): void => {
+  const skinUiCss = skinKey + '/skin.css';
+  const css = EditorManager.resources.get(skinUiCss);
+
+  if (Type.isString(css)) {
+    return loadRawCss(editor, skinUiCss, css, editor.ui.styleSheetLoader);
+  }
+};
+
+const loadShadowDomRawUiSkins = (editor: Editor, skinKey: string): void => {
+  const isInShadowRoot = SugarShadowDom.isInShadowRoot(SugarElement.fromDom(editor.getElement()));
+  if (isInShadowRoot) {
+    const shadowDomSkinCss = skinKey + '/skin.shadowdom.css';
+    const css = EditorManager.resources.get(shadowDomSkinCss);
+
+    if (Type.isString(css)) {
+      return loadRawCss(editor, shadowDomSkinCss, css, DOMUtils.DOM.styleSheetLoader);
+    }
+  }
+};
+
+const loadKeySkin = (isInline: boolean, editor: Editor): Promise<void | [void, void]> => {
+  const skinKey = Options.getSkinKey(editor);
+
+  if (skinKey) {
+    // eslint-disable-next-line no-debugger
+    debugger;
+    const css = EditorManager.resources.get(skinKey + (isInline ? '/content.inline' : '/content') + '.css');
+
+    if (Type.isString(css)) {
+      loadRawCss(editor, skinKey, css, editor.ui.styleSheetLoader);
+    }
+  }
+
+  // In Modern Inline, this is explicitly called in editor.on('focus', ...) as well as in render().
+  // Seems to work without, but adding a note in case things break later
+  if (!Options.isSkinDisabled(editor) && Type.isString(skinKey)) {
+    loadRawUiSkins(editor, skinKey);
+    loadShadowDomRawUiSkins(editor, skinKey);
+  }
+
+  return Promise.resolve();
+};
+
+const loadSkin = (isInline: boolean, editor: Editor): Promise<void> => {
+  return Promise.all([
```

Comment:
This will run the promises in parallel, which doesn't seem like the right answer. IMO it should check whether we want to load from url or key, and then do the one thing it's supposed to.

Reply:
We could do that, if you prefer, but I had some concerns that people might always want to use the normal url one, and the other was a suppliment. So just using one wasn't ideal. But I might have been wrong.

Reply:
I think loading the resource as a string should always take priority over loading from the URL, even if the key is a URL.